File: Back/src/ProEventos.Domain/Event.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProEventos.Domain;

[Table("Events")]
public class Event
{
    [Key] public int Id { get; set; }
    public string Local { get; set; }
    public DateTime Date { get; set; }
    [Required] public string Theme { get; set; }
    public int NumberOfPeoples { get; set; }
    public string Base64 { get; set; }
    public string Phone { get; set; }
    public string Email { get; set; }
    public IEnumerable<Batch> Batches { get; set; }
    public IEnumerable<SocialMedia> SocialMedias { get; set; }
    public Speaker Speaker { get; set; }
    public int SpeakerId { get; set; }
}

File: Front/Proeventos-app/src/app/models/Event.ts
import {Lot} from "./Lot";
import {SocialMedia} from "./SocialMedia";
import {EventSpeaker} from "./EventSpeaker";



export interface Event {
  Id: number;
  Local: string;
  Date?: Date;
  Theme: string;
  NumberOfPeoples: number;
  imgURL: string;
  phone: string;
  email: string;
  Lots: Lot[];
  SocialMedias: SocialMedia[];
  SpeakersEvent: EventSpeaker[];
}
